Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2/Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "ROS2/ROS2GemUtilities.h"
+#include "ROS2/Utilities/ROS2Names.h"
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+namespace ROS2
+{
+    namespace Internal
+    {
+        AZ::TransformInterface* GetEntityTransformInterface(const AZ::Entity* entity)
+        {
+            // TODO - instead, use EditorFrameComponent to handle Editor-context queries and here only use the "Game" version
```

Comment:
TODO comments should be made into a GHI or removed if not needed instead of propagating in the code base: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#todo-comments

TODO comments doesn't enforce or provide a responsibility for anyone to implement their functionality.
The full context of what needs to be done is only known by the person who wrote the comment and they might not be around to implement what the TODO requires or they might forget.
Having an issue allows the developer to provide more context as to what needs to be done and allows tracking of the work in project management tools